src/js/models/v2/aboutTheData/BaseReportingPeriodRow.js

```diff
@@ -14,8 +14,13 @@ BaseReportingPeriodRow.populate = function populate(data) {
     this.fiscalYear = parseInt(data.fiscal_year, 10) || 0;
     this.fiscalPeriod = parseInt(data.fiscal_period, 10) || 0;
     this.reportingPeriod = `FY ${this.fiscalYear}: ${getPeriodWithTitleById(`${this.fiscalPeriod}`).title}`;
-    this._percentOfBudget = data.percent_of_total_budgetary_resources || 0;
-    this.percentOfBudget = `${formatNumberWithPrecision(this._percentOfBudget, 2)}%`;
+    this._percentOfBudget = data.percent_of_total_budgetary_resources;
+    if (!this._percentOfBudget && typeof this._percentOfBudget !== 'number') {
```

Comment:
Could we put this into a nice reusable helper fn somewhere like we do with calculate percentage?

Reply:
Put what into a nice reusable helper? 

1. This `!this._percentOfBudget && typeof this._percentOfBudget !== 'number'`

2. Or this 
```
if (!this._percentOfBudget && typeof this._percentOfBudget !== 'number') {
     this.percentOfBudget = '--';
    }
    else {
        this.percentOfBudget = `${formatNumberWithPrecision(this._percentOfBudget, 2)}%`;
    }
```

Reply:
Just the whole logic around the ultimate value of `_percentOfBudget`

Reply:
pushed an update, this should be easier to follow with no need for a helper.